src/main/java/com/spotify/annoy/ANNIndex.java

```diff
@@ -146,63 +166,88 @@ public int compareTo(final PQEntry o) {
     Set<Integer> nearestNeighbors = new HashSet<Integer>();
     while (nearestNeighbors.size() < roots.size() * nResults && !pq.isEmpty()) {
       PQEntry top = pq.poll();
-      int n = top.node;
-      int nDescendants = annBuf.getInt(n);
-      getNodeVector(n, v);
-      if(Arrays.equals(v, ANNIndex.zeros))
-          continue;
+      int topNodeOffset = top.nodeOffset;
+      int nDescendants = annBuf.getInt(topNodeOffset);
+      float[] v = getNodeVector(topNodeOffset);
+      if (isZeroVec(v))
+        continue;
       if (nDescendants == 1) {  // n_descendants
         // FIXME: does this ever happen?
-        nearestNeighbors.add(n / nodeSize);
+        nearestNeighbors.add(topNodeOffset / nodeSize);
       } else if (nDescendants <= minLeafSize) {
         for (int i = 0; i < nDescendants; i++) {
-          int j = annBuf.getInt(4 + n + i * kFloatSize);
+          int j = annBuf.getInt(topNodeOffset +
+				((indexType == IndexType.ANGULAR) ? 4 : 8) +
+				i * INT_SIZE);
           nearestNeighbors.add(j);
         }
       } else {
-        float margin = cosineMargin(v, queryVector);
-        int lChild = nodeSize * annBuf.getInt(n + 4);
-        int rChild = nodeSize * annBuf.getInt(n + 8);
+        float margin = (indexType == IndexType.ANGULAR) ?
+	  cosineMargin(v, queryVector) :
+	  euclideanMargin(v, queryVector, getNodeBias(topNodeOffset));
+	int childrenMemOffset = topNodeOffset + (indexType == IndexType.ANGULAR ? 4 : 8);
```

Comment:
I am truly deeply sorry. I regret my actions and will do my best to improve in the future :)
I'll run a final `M-x untabify` after fixing all the indentations